### ISSUE TITLE
[BUG] Allow import of signed PDFs (by default)

### ISSUE THREAD

**Issue by chrisblech** [paperless-ngx/paperless-ngx#7383]
### Description

Preface:
Basically, there is already a (similar, now closed: #4047) issue on this topic. It also mentions a workaround that works. In my opinion, however, this workaround should not only be reserved for special cases, but should be included in the standard behavior of Paperless.
In view of the fact that more and more PDFs will contain a digital signature in the future, the problem described here will become increasingly relevant.

In a normal Paperless installation, the import of a digitally signed PDF is aborted with an error message: `DigitalSignatureError: Input PDF has a digital signature. OCR would alter the document, invalidating the signature.
`
The cause lies in the "OCRmyPDF" module used by Paperless for OCR, which by default prefers not to use OCR in order to preserve the signature.
In the context of Paperless, however, this decision is to be evaluated differently: Since Paperless additionally saves the original PDF (with the digital signature) anyway, the loss of the digital signature in a copy of the document (in favor of OCR in this version) is not a problem.
Paperless should therefore instruct the OCRmyPDF module by default not to take any existing digital signature into account during OCR. This is easily possible with a command line parameter.

@tomfitzhenry recommends in https://github.com/paperless-ngx/paperless-ngx/discussions/4047#discussioncomment-7019544 to set ENV `PAPERLESS_OCR_USER_ARGS` to `{"invalidate_digital_signatures": true}`, but this seems (as the further discussion there shows, due to the required escaping of quotation marks) sometimes very difficult in the different Docker configurations.

So my suggestion is to set this value as the default, unless the user explicitly sets it differently.

### Steps to reproduce

1. Spin up a fresh instance of paperless
2. upload a digitally signed PDF

### Webserver logs

```bash
[2024-08-02 18:51:17,841] [ERROR] [paperless.tasks] ConsumeTaskPlugin failed: Antrag.pdf: Error occurred while consuming document Antrag.pdf: DigitalSignatureError: Input PDF has a digital signature. OCR would alter the document,

invalidating the signature.

Traceback (most recent call last):

  File "/usr/src/paperless/src/paperless_tesseract/parsers.py", line 368, in parse

    ocrmypdf.ocr(**args)

  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/api.py", line 375, in ocr

    return run_pipeline(options=options, plugin_manager=plugin_manager)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/_pipelines/ocr.py", line 225, in run_pipeline

    return _run_pipeline(options, plugin_manager)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/_pipelines/ocr.py", line 189, in _run_pipeline

    validate_pdfinfo_options(context)

  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/_pipeline.py", line 202, in validate_pdfinfo_options

    raise DigitalSignatureError()

ocrmypdf.exceptions.DigitalSignatureError: Input PDF has a digital signature. OCR would alter the document,

invalidating the signature.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):

  File "/usr/src/paperless/src/paperless_tesseract/parsers.py", line 419, in parse

    ocrmypdf.ocr(**args)

  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/api.py", line 375, in ocr

    return run_pipeline(options=options, plugin_manager=plugin_manager)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/_pipelines/ocr.py", line 225, in run_pipeline

    return _run_pipeline(options, plugin_manager)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/_pipelines/ocr.py", line 189, in _run_pipeline

    validate_pdfinfo_options(context)

  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/_pipeline.py", line 202, in validate_pdfinfo_options

    raise DigitalSignatureError()

ocrmypdf.exceptions.DigitalSignatureError: Input PDF has a digital signature. OCR would alter the document,

invalidating the signature.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):

  File "/usr/local/lib/python3.11/site-packages/asgiref/sync.py", line 327, in main_wrap

    raise exc_info[1]

  File "/usr/src/paperless/src/documents/consumer.py", line 582, in run

    document_parser.parse(self.working_copy, mime_type, self.filename)

  File "/usr/src/paperless/src/paperless_tesseract/parsers.py", line 431, in parse

    raise ParseError(f"{e.__class__.__name__}: {e!s}") from e

documents.parsers.ParseError: DigitalSignatureError: Input PDF has a digital signature. OCR would alter the document,

invalidating the signature.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):

  File "/usr/src/paperless/src/documents/tasks.py", line 151, in consume_file

    msg = plugin.run()

          ^^^^^^^^^^^^

  File "/usr/src/paperless/src/documents/consumer.py", line 613, in run

    self._fail(

  File "/usr/src/paperless/src/documents/consumer.py", line 302, in _fail

    raise ConsumerError(f"{self.filename}: {log_message or message}") from exception

documents.consumer.ConsumerError: Antrag.pdf: Error occurred while consuming document Antrag.pdf: DigitalSignatureError: Input PDF has a digital signature. OCR would alter the document, invalidating the signature.
```

### Paperless-ngx version

2.11.1

### Host OS

Debian 12 Bookworm

### Installation method

Docker - official image

### Browser

Chrome

### Please confirm the following

- [X] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [X] I have already searched for relevant existing issues and discussions before opening this report.
- [X] I have updated the title field above with a concise description.

**Comment by shamoon** [comment]
This is not a bug report it’s a change request. Please don’t do that.

There are multiple ways to set that parameter and using environment is not a requirement. Lots of people care about preserving digital signatures

**Comment by chrisblech** [comment]
Thanks for your feedback - which unfortunately I can't agree with.

> This is not a bug report it’s a change request. Please don’t do that.

IMHO, most people use Paperless to archive PDFs and expect to be able to find them again via content search. 

When importing a signed PDF in the current Paperless version, an error message appears and the PDF is not saved at all (so this is a bug).

> Lots of people care about preserving digital signatures

With the proposed bugfix, importing a signed PDF will work as desired again:
- the PDF is archived without an error message
- the full text search works, as OCR was performed on the copy
- the original (with digital signature) is available via "Download original" 

> There are multiple ways to set that parameter and using environment is not a requirement. 

Sorry if my proposed fix was unclear. I would suggest to always use `{"invalidate_digital_signatures": true}` as parameter for OCR, when `OCR_USER_ARGS` is not set explicitely otherwise.

**Comment by shamoon** [comment]
> When importing a signed PDF in the current Paperless version, an error message appears and the PDF is not saved at all (so this is a bug).

It tells you why it wasn’t imported. Talk to the people who live in countries with strict regulation (Germans) who use this software, invalidating a digital signature makes the document no longer legally “unmodified” so no we’re not going to that without the user making an explicit choice.

This has been the way it’s worked for a while now, reporting that this is somehow suddenly a bug comes across as petulant and is a waste of our time dealing with a bogus bug report. Again, please don’t do that.

If you want to discuss other ways to handle this, make it easier or something then this is not the appropriate avenue to do that.

**Comment by DenisMir** [comment]
So what is the correct way to handle that? Print it out? What is the purpose then when Paperless can't handle signed documents.

**Comment by thiswillbeyourgithub** [comment]
Given that the goal of PDF signing in the first place is to forbid file alteration, I don't understand why paperless can't store it as "original".

Paperless already handles having both an "original PDF" and an "archived PDF". So why isn't it possible to have the original be signed?

The german users in your example, if asked to send the signed PDF could just click on "download the original PDF" right?

But also, if the laws are so strict that you are expected to keep many original documents, that means you are among the users that have the most need for an archival solution that accepts them. Currently I'm guessing german users rely on paperless for 95% of their PDF (including very important one) but the ones that are signed have to be either stored separately or unsigned then saved to paperless and elsewhere. This is increasing both the burden on the user AND the risk of losing them.

So to summarize I think I understand where this policy comes from but the effect might be that users that need paperless the most can't use it for what's important.

**Comment by shamoon** [comment]
Paperless does store the original signed PDF, that setting is only for the archive version.
